cli/internal/oauth/util/browser.go

```diff
@@ -0,0 +1,37 @@
+package util
+
+import (
+	"errors"
+	"os/exec"
+	"runtime"
+	"strings"
+)
+
+// OpenBrowser opens the specified URL in a browser based on OS.
+func OpenBrowser(url string) error {
+	switch runtime.GOOS {
+	case "linux":
+		return openBrowserLinux(url)
+	case "windows":
+		return exec.Command("rundll32", "url.dll,FileProtocolHandler", url).Start()
+	case "darwin":
+		return exec.Command("open", url).Start()
+	default:
+		return errors.New("unsupported platform")
+	}
+}
+
+func openBrowserLinux(url string) error {
+	providers := []string{"wslview", "xdg-open", "x-www-browser", "www-browser"}
+
+	// There are multiple possible providers to open a browser on linux
```

Comment:
Oh! Or was this copied from there? I just noticed a very similar comment in that code; https://github.com/pkg/browser/blob/5ac0b6a4141c771a0d3a081c36b87c977cf9c7db/browser_linux.go#L11


If we copied, we must probably include license and copyright from there, otherwise we're violating the license; https://github.com/pkg/browser/blob/5ac0b6a4141c771a0d3a081c36b87c977cf9c7db/LICENSE#L10


(but maybe using the dependency would be a solution for that if we can use it as-is)

Reply:
For sure need to include the copyright notice, but I wonder if just using the dependency would work for [WSL](https://github.com/docker/cli/pull/5244#discussion_r1677848277) (no `wslview` in that code). OTOH, maybe it doesn't need to, according to this [post](https://askubuntu.com/questions/1409481/why-does-xdg-open-use-wslview) `xdg-open` would probably end up calling `wslview` under the hood.

Reply:
I'll push a version vendoring `pkg/browser`

Reply:
Thanks! Yeah, you know I usually try to reduce dependencies, but I know it's already in our tree indirectly (in some cases), so using the dependency instead of maintaining our own fork of the code could be a good thing (assuming that works).

We can still decide to maintain a fork if there's a need (well, or contribute to upstream if it's something that would be generally useful to improve in that module)